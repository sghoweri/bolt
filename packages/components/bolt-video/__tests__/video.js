import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
} from '../../../../packages-extras/testing/testing-helpers';
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../video.schema.yml'));

describe('<bolt-video> Component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  test('<bolt-video> compiles', async () => {
    const results = await render('@bolt-components-video/video.twig', {
      videoId: '4892122320001',
      accountId: '1900410236',
      playerId: 'r1CAdLzTW',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('<bolt-video> compiles with a meta description', async () => {
    const results = await render('@bolt-components-video/video.twig', {
      videoId: '4892122320001',
      accountId: '1900410236',
      playerId: 'r1CAdLzTW',
      showMeta: true,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('<bolt-video> compiles with a meta description + meta title', async () => {
    const results = await render('@bolt-components-video/video.twig', {
      videoId: '4892122320001',
      accountId: '1900410236',
      playerId: 'r1CAdLzTW',
      showMeta: true,
      showMetaTitle: true,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('<bolt-video> compiles with a different `video-id`', async () => {
    const results = await render('@bolt-components-video/video.twig', {
      videoId: '5609376179001',
      accountId: '1900410236',
      playerId: 'r1CAdLzTW',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // tag.enum.forEach(async tagChoice => {
  //   test(`button tag: ${tagChoice}`, async () => {
  //     const results = await render('@bolt-components-button/button.twig', {
  //       text: 'Hello World',
  //       tag: tagChoice,
  //     });
  //     // console.log({ results });
  //     expect(results.ok).toBe(true);
  //     expect(results.html).toMatchSnapshot();
  //   });
  // });
});
