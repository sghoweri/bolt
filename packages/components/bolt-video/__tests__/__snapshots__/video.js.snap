// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-video> Component <bolt-video> compiles 1`] = `
<bolt-ratio ratio="16/9">
  <ssr-keep for="bolt-ratio"
            class="c-bolt-ratio"
            style="padding-bottom: calc((9 / 16) * 100%); --aspect-ratio: 1.7777777777778;"
  >
    <bolt-video class="js-bolt-video-uuid--12345"
                video-id="4892122320001"
                account-id="1900410236"
                show-meta
                player-id="r1CAdLzTW"
                controls
                share-description="Share This Video"
    >
    </bolt-video>
  </ssr-keep>
</bolt-ratio>
`;

exports[`<bolt-video> Component <bolt-video> compiles with a different \`video-id\` 1`] = `
<bolt-ratio ratio="16/9">
  <ssr-keep for="bolt-ratio"
            class="c-bolt-ratio"
            style="padding-bottom: calc((9 / 16) * 100%); --aspect-ratio: 1.7777777777778;"
  >
    <bolt-video class="js-bolt-video-uuid--12345"
                video-id="5609376179001"
                account-id="1900410236"
                show-meta
                player-id="r1CAdLzTW"
                controls
                share-description="Share This Video"
    >
    </bolt-video>
  </ssr-keep>
</bolt-ratio>
`;

exports[`<bolt-video> Component <bolt-video> compiles with a meta description + meta title 1`] = `
<bolt-ratio ratio="16/9">
  <ssr-keep for="bolt-ratio"
            class="c-bolt-ratio"
            style="padding-bottom: calc((9 / 16) * 100%); --aspect-ratio: 1.7777777777778;"
  >
    <bolt-video class="js-bolt-video-uuid--12345"
                video-id="4892122320001"
                account-id="1900410236"
                show-meta
                show-meta-title
                player-id="r1CAdLzTW"
                controls
                share-description="Share This Video"
    >
    </bolt-video>
  </ssr-keep>
</bolt-ratio>
`;

exports[`<bolt-video> Component <bolt-video> compiles with a meta description 1`] = `
<bolt-ratio ratio="16/9">
  <ssr-keep for="bolt-ratio"
            class="c-bolt-ratio"
            style="padding-bottom: calc((9 / 16) * 100%); --aspect-ratio: 1.7777777777778;"
  >
    <bolt-video class="js-bolt-video-uuid--12345"
                video-id="4892122320001"
                account-id="1900410236"
                show-meta
                player-id="r1CAdLzTW"
                controls
                share-description="Share This Video"
    >
    </bolt-video>
  </ssr-keep>
</bolt-ratio>
`;
